src/libaktualizr/primary/initializer.cc

```diff
@@ -205,7 +205,7 @@ InitRetCode Initializer::initEcuRegister() {
 Initializer::Initializer(
     const ProvisionConfig& config_in, std::shared_ptr<INvStorage> storage_in,
     std::shared_ptr<HttpInterface> http_client_in, KeyManager& keys_in,
-    const std::map<Uptane::EcuSerial, std::shared_ptr<Uptane::SecondaryInterface> >& secondary_info_in)
+    const std::map<Uptane::EcuSerial, std::shared_ptr<Uptane::SecondaryInterface> >& secondary_info_in, bool lite_mode)
```

Comment:
I don't particularly like punching holes in our initialization logic, but I haven't yet thought of a better way that isn't just duplicating code. But can we at least change the name? `lite_mode` seems rather hard to understand without already knowing the context. Perhaps `tls_only` would be better? I do like that it defaults to false, though!

Reply:
I'm a bit uneasy about this as well. Adding conditional logic into a security related flow could cause a subtle bug some day. I think there are two other options here:

1) Keep this patch out-of-tree.
Its kind of specific to something we are doing. I can understand if this isn't interesting to you guys.

2) Have aktualizr-lite use the SotatUptaneClient constructor rather than calling `newDefaultClient`
If we changed things to this, then I could actually just call what I needed directly, ` keys_.copyCertsToCurl(*http_client_);`

I haven't tried out option 2 but I think that might work better here and not leak details of the lite stuff into sotauptaneclient. Sound reasonable?